econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -138,25 +138,30 @@ def _first_stage_reg(X, y, *, automl=True, random_state=None, verbose=0):
                                  scoring='r2',
                                  verbose=verbose)
         best_est = model.fit(X, y).best_estimator_
-        if isinstance(best_est, Pipeline):
-            return make_pipeline(StandardScaler(), Lasso(alpha=best_est.steps[1][1].alpha_, random_state=random_state))
+        if isinstance(best_est, LassoCV):
+            return Lasso(alpha=best_est.alpha_, random_state=random_state)
         else:
             return best_est
     else:
-        model = make_pipeline(StandardScaler(), LassoCV(cv=5, random_state=random_state)).fit(X, y)
-        return make_pipeline(StandardScaler(), Lasso(alpha=model.steps[1][1].alpha_, random_state=random_state))
+        model = LassoCV(cv=5, random_state=random_state).fit(X, y)
+        return Lasso(alpha=model.alpha_, random_state=random_state)
 
 
 def _first_stage_clf(X, y, *, make_regressor=False, automl=True, min_count=None, random_state=None, verbose=0):
     if min_count is None:
         min_count = _CAT_LIMIT  # we have at least this many instances
     if automl:
-        model = GridSearchCVList([make_pipeline(StandardScaler(), LogisticRegression(max_iter=1000,
-                                                                                     random_state=random_state)),
+        # use same Cs as would be used by default by LogisticRegressionCV
+        # NOTE: we don't use LogisticRegressionCV inside the grid search because of the nested stratification
+        #       which could affect how many times each distinct Y value needs to be present in the data
+        cs = LogisticRegressionCV().fit(np.empty((10, 1)), 5 * [1] + 5 * [0]).Cs_
```

Comment:
To avoid cases where sklearn switches to an adaptive setting of the Cs array based on the values of the input variables, which in this case will be weird, it might be better to just use an explicit log-space array which is implemented internally. We could in principle hard code the same array when we use LogisticRegressionCV() for consistency.

Reply:
Okay, I'll use a fixed log-space array both here and above.